draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -69,74 +70,60 @@ informative:
-                             ^ |
-                             | |
-  +--------------------------|-|--------------+
-  |Cluster                   | | OAuth        |
-  |                          | | Exchange     |
-  | +---------------+        | | to obtain    |
-  | |               |        | | access token |
-  | | Control Plane |        | | using        |
-  | |               |        | | Service      |
-  | +---------------+        | | Account      |
-  |         ^|               | | Token        |
-  |         ||               | v        +-----+
-  |         ||            +----------+  |
-  |         ||            |          |+ |        +----------+
-  |  Obtain ||            | Workload || |        |          |
-  |  Service||            |          ||<-------->| Resource |
-  |  Account||            +----------+| | Access | Server   |
-  |  Token  ||             +----------+ | Token  |          |
-  |         ||                  ^       |        +----------+
-  |         ||    Start Workload:       |
-  |         ||     with Service :       |
-  |         ||    Account Token :       |
-  |         ||                  v       |
-  |         ||              +-------+   |
-  |         |+------------->|       |   |
-  |         +---------------| Agent |   |
-  |                         |       |   |
-  |                         +-------+   |
-  |                                     |
-  +-------------------------------------+
+┌──────────────────────────────────────────────────────────┐
+│                           External Authorization Domain  │
+│                                                          │
+│ ┌─────────────────────────┐     ┌────────────────────┐   │
+│ │                         │     │                    │   │
+│ │   Authorization Server  │     │  Resource Server   │   │
+│ │                         │     │                    │   │
+│ └────────────┬────────────┘     └────────────▲───────┘   │
+│              │                               │           │
+└──────────────┼───────────────────────────────┼───────────┘
+               │                               │
+     2) present assertion                 3) access
+               │                               │
+               │     ┌─────────────────────────┘
+               │     │
+┌──────────────┼─────┼─────────────────────────────────────┐
+│              │     │                  Workload Platform  │
+│              │     │                                     │
+│  ┌───────────┴─────┴───┐           ┌──────────────────┐  │
+│  │                     │           │ Credential       │  │
+│  │  Workload           ├───────────► issued by        │  │
+│  │                     │  1) get   │ Platform         │  │
+│  └─────────────────────┘           └──────────────────┘  │
+│                                                          │
+│                                                          │
+└──────────────────────────────────────────────────────────┘
```

Comment:
I Like the new diagram, Its much more agnostic and to k8s, but still shows the approach clearly.